### PR TITLE
refactor: convert to kotlin

### PR DESCRIPTION
**Pull request by Charm-Xin**

### CODE REVIEW

File: oss-common/src/main/java/dev/ocpd/oss/FileStore.kt [hunk 1]
```diff
@@ -19,63 +16,62 @@ public interface FileStore {
...
+    fun upload(key: String, content: ByteArray)
 
     /**
      * Upload a file to the file store.
      *
      * @param key The key of the file to be uploaded.
      * @param is  The input stream of the file to be uploaded, which will not be closed.
      */
-    void upload(String key, InputStream is);
+    fun upload(key: String, `is`: InputStream)
```
Comment (EggOxygen) [inline oss-common/src/main/java/dev/ocpd/oss/FileStore.kt:43]: inputStream

File: oss-gcs/src/main/java/dev/ocpd/oss/config/GcsFileStoreAutoConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@AutoConfiguration
+@ConditionalOnClass(Storage::class)
+@ConditionalOnProperty(prefix = "oss", name = ["provider"], havingValue = "gcs")
+@EnableConfigurationProperties(GcsFileStoreProperties::class)
+class GcsFileStoreAutoConfiguration(
+    private val properties: GcsFileStoreProperties,
+    private val storage: Storage
+) {
+    private val log = LoggerFactory.getLogger(GcsFileStoreAutoConfiguration::class.java)
+
```
Comment (EggOxygen) [inline oss-gcs/src/main/java/dev/ocpd/oss/config/GcsFileStoreAutoConfiguration.kt:23]: private val log by slf4j

File: oss-gcs/src/main/java/dev/ocpd/oss/GcsFileStore.kt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.concurrent.TimeUnit
+import java.util.stream.Stream
+import java.util.stream.StreamSupport
+
+class GcsFileStore(
+    private val storage: Storage,
+    private val bucket: String
+) : FileStore {
+
+    private val log = LoggerFactory.getLogger(GcsFileStore::class.java)
```
Comment (EggOxygen) [inline oss-gcs/src/main/java/dev/ocpd/oss/GcsFileStore.kt:24]: private val log by slf4j

File: oss-s3/src/main/java/dev/ocpd/oss/AwsS3FileStore.kt [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import java.util.stream.Stream
+import java.util.stream.StreamSupport
+
+class AwsS3FileStore(
+    private val client: S3Client,
+    private val s3Presigner: S3Presigner,
+    private val bucket: String
+) : FileStore {
+
+    private val log = LoggerFactory.getLogger(AwsS3FileStore::class.java)
```
Comment (EggOxygen) [inline oss-s3/src/main/java/dev/ocpd/oss/AwsS3FileStore.kt:25]: private val log by slf4j

**Review by EggOxygen — CHANGES_REQUESTED** [review]

File: oss-test/src/main/java/dev/ocpd/oss/FileStoreTest.kt [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    @Autowired
+    private lateinit var fileStore: FileStore
+
+    @AfterEach
+    fun clear() {
+        fileStore.delete(TEST_FILE_KEY)
+    }
+
+    @Test
+    @Throws(IOException::class)
```
Comment (EggOxygen) [inline oss-test/src/main/java/dev/ocpd/oss/FileStoreTest.kt:36]: Don't need this i guess?

File: oss-s3/src/test/java/dev/ocpd/oss/AwsS3FileStoreTest.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ContextConfiguration
+
+@SpringBootTest(classes = [AwsS3FileStoreAutoConfiguration::class])
+@ContextConfiguration(initializers = [AwsPropertiesEnvInitializer::class])
+object AwsS3FileStoreTest : FileStoreTest() {
+
+    @JvmStatic
```
Comment (EggOxygen) [inline oss-s3/src/test/java/dev/ocpd/oss/AwsS3FileStoreTest.kt:16]: use `@TestInstance(PER_CLASS)`

**Review by EggOxygen — CHANGES_REQUESTED** [review]

File: oss-common/src/main/java/dev/ocpd/oss/FileStore.kt (hunk 1 above)
Comment (unlimitedsola) [reply oss-common/src/main/java/dev/ocpd/oss/FileStore.kt:43]: or `ins` would also work

File: oss-s3/src/test/java/dev/ocpd/oss/AwsS3FileStoreTest.kt (hunk 6 above)
Comment (Charm-Xin) [reply oss-s3/src/test/java/dev/ocpd/oss/AwsS3FileStoreTest.kt:16]: Localstack docker beans do not work with `PER_CLASS` JUnit Lifecycle.

see
https://github.com/localstack/localstack-java-utils/issues/104

File: oss-test/src/main/java/dev/ocpd/oss/FileStoreTest.kt [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+
+abstract class FileStoreTest {
+
+    companion object {
+        private const val TEST_FILE_KEY_PREFIX = "oss-test-images"
+        private const val TEST_FILE_KEY = "$TEST_FILE_KEY_PREFIX/binglogo.png"
+        private val TEST_FILE_URL: URL = try {
```
Comment (EggOxygen) [inline oss-test/src/main/java/dev/ocpd/oss/FileStoreTest.kt:19]: We don't need to check for invalid URL here, I guess? This is a preset value and will never be invalidated.

File: oss-gcs/src/main/java/dev/ocpd/oss/GcsFileStore.kt [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+    override fun downloadAsBytes(key: String): ByteArray {
+        val fileContent = storage.readAllBytes(bucket, key)
+
+        log.debug { "Downloaded file with key: $key" }
+        return fileContent
+    }
+
+    override fun downloadTo(key: String, outputStream: OutputStream) {
+        val blob = storage[bucket, key]
```
Comment (unlimitedsola) [inline oss-gcs/src/main/java/dev/ocpd/oss/GcsFileStore.kt:62]:
```suggestion
        val blob = storage[bucket, key] ?: throw IllegalArgumentException("File does not exist: $key")
```

**Review by unlimitedsola — APPROVED** [review]

**Pull request merged**